scripts/nbconverted/integrate-compression-hetnet.py

```diff
@@ -0,0 +1,348 @@
+)
+
+coding_genes = set(gene_df['entrez_gene_id'].astype(int))
+print(gene_df.shape)
+gene_df.head(2)
+
+
+# ## Add genes as nodes to the graph
+# 
+# Use the gene-symbol identifier for easier interpretation
+
+# In[6]:
+
+
+get_ipython().run_cell_magic('time', '', "for i, row in gene_df.iterrows():\n    # Build dictionary of descriptive elements for each gene\n    meta_data = {\n        'description': row['description'],\n        'source': 'Entrez Gene',\n        'url': 'http://identifiers.org/ncbigene/{}'.format(row['entrez_gene_id']),\n        'license': 'CC0 1.0',\n    }\n    \n    if pd.notnull(row['chromosome']):\n        meta_data['chromosome'] = row['chromosome']\n\n    # Add genes to graph\n    graph.add_node(kind='Gene', identifier=int(row['entrez_gene_id']), name=row['symbol'],\n                   data=meta_data)")
+
+
+# In[7]:
+
+
+# Load gene updater
+url = 'https://raw.githubusercontent.com/cognoma/genes/{}/data/updater.tsv'.format(commit)
+updater_df = pd.read_table(url)
+old_to_new_entrez = dict(zip(updater_df.old_entrez_gene_id,
+                             updater_df.new_entrez_gene_id))
+
+
+# ## Add gene set nodes and associated genes as edges
+# 
+# Add each MSigDB collection as distinct nodes with a `participates` edge for representative gene sets and corresponding membership.
+
+# In[8]:
+
+
+def add_node_to_graph(current_graph, collection_file, collection_kind,
+                      collection_source, gene_list, min_geneset_size=4,
+                      max_geneset_size=1000, license='CC BY 4.0'):
+    """
+    Add nodes and edges to current graph based on geneset memembership of collection
+    
+    Arguments:
+    current_graph - a hetnet object to add node-edge info to
+    collection_file - location of msigdb file
+    collection_kind - the kind of node already initialized in the graph
+    collection_source - alternative ID for collection
+    gene_list - a list of genes to consider when building the graph
+    min_geneset_size - filter out a given gene set if it has fewer genes
+    max_geneset_size - filter out a given gene set if it has more genes
+    license - given license associated with node
+    
+    Output:
+    Adds to current graph; Returns the amount of filtered genesets
+    """
+    
+    # Build meta data dictionary to store node info
+    meta_data = {'license': license, 'source': collection_source}
+    
+    # Open the .gmt file and process each geneset
+    filtered_genesets = []
+    with open(collection_file, 'r') as collection_fh:
+        collection_reader = csv.reader(collection_fh, delimiter='\t')
+
+        for row in collection_reader:
+            # Get geneset and and metadata info
+            geneset_name = row[0]
+            meta_data['url'] = row[1]
+            
+            # Process geneset membership
+            genes = row[2:]
+
+            # Update entrez_gene_id
+            genes = [old_to_new_entrez[x] if x in old_to_new_entrez else x for x in genes]
```

Comment:
I think you want to make to use a set, in the rare case an old entrez gene ID maps to a new ID already in the set.